tests/integ/test_udf.py

```diff
@@ -1548,7 +1582,166 @@ def get_numpy_pandas_version() -> str:
+):
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_requirements(test_files.test_unsupported_requirements_file)
+        assert set(session.get_packages().keys()) == {
+            "scipy",
+            "numpy",
+            "matplotlib",
+            "pyyaml",
+        }
+
+        session.add_requirements(test_files.test_unsupported_requirements_file)
+        assert set(session.get_packages().keys()) == {
+            "scipy",
+            "numpy",
+            "matplotlib",
+            "pyyaml",
+        }
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Subprocess calls are not allowed within stored procedures",
+)
+def test_add_packages_should_fail_if_dependency_package_already_added(session):
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_packages(["scikit-learn==1.2.0"])
+        with pytest.raises(ValueError) as ex_info:
+            session.add_packages("sktime")
+        assert "Cannot add dependency package" in str(ex_info)
+
+
+@pytest.mark.skipif(
+    IS_IN_STORED_PROC,
+    reason="Subprocess calls are not allowed within stored procedures",
+)
+def test_add_requirements_unsupported(session, resources_path):
+    test_files = TestFiles(resources_path)
+
+    with patch.object(session, "_is_anaconda_terms_acknowledged", lambda: True):
+        session.add_requirements(test_files.test_unsupported_requirements_file)
+        # Once scikit-fuzzy is supported, this test will break; change the test to a different unsupported module
+        assert set(session.get_packages().keys()) == {
+            "matplotlib",
+            "pyyaml",
+            "scipy",
+            "numpy",
+        }
+
+    udf_name = Utils.random_name_for_temp_object(TempObjectType.FUNCTION)
+
+    @udf(name=udf_name)
+    def run_scikit_fuzzy() -> str:
+        import numpy as np
+        import skfuzzy as fuzz
+        from skfuzzy import control as ctrl
+
+        # Create fuzzy variables
+        temperature = ctrl.Antecedent(np.arange(0, 101, 1), "temperature")
+        humidity = ctrl.Antecedent(np.arange(0, 101, 1), "humidity")
+        fan_speed = ctrl.Consequent(np.arange(0, 101, 1), "fan_speed")
+
+        # Define fuzzy membership functions
+        temperature["cold"] = fuzz.trimf(temperature.universe, [0, 0, 50])
+        temperature["warm"] = fuzz.trimf(temperature.universe, [0, 50, 100])
+        humidity["dry"] = fuzz.trimf(humidity.universe, [0, 0, 50])
+        humidity["moist"] = fuzz.trimf(humidity.universe, [0, 50, 100])
+        fan_speed["low"] = fuzz.trimf(fan_speed.universe, [0, 0, 50])
+        fan_speed["high"] = fuzz.trimf(fan_speed.universe, [0, 50, 100])
+
+        # Define fuzzy rules
+        rule1 = ctrl.Rule(temperature["cold"] & humidity["dry"], fan_speed["low"])
+        rule2 = ctrl.Rule(temperature["warm"] & humidity["moist"], fan_speed["high"])
```

Comment:
same comment about complexity of the udf. It is better if we have something simple.

Reply:
Not sure if this changes things but the code here runs fast.

Reply:
The reason I am in favor of a simple code here is so that if things break and we need to debug, it is easier to debug a simpler code rather than a complex code.

Reply:
Thx, updated to a simpler test.